### PR TITLE
waybar: accept null as package

### PR DESCRIPTION
**Pull request by thrix**
### Description

I use `home-manager` in Fedora Silverblue toolbox to configure my home.
I do not need waybar package installed, it is installed by my OS.

Seem this could be nicely achieved by:

    package = null;

Docs currently say:

    Waybar package to use. Set to `null` to install the default package.

What is a bit misleading, you cannot set it to really to a `null` value, you can omit it.

This MR provides a way to provide a null value to skip the package installation.

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all` or `nix develop --ignore-environment .#all` using Flakes.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

### CODE REVIEW

File: modules/programs/waybar.nix [hunk 1]
```diff
@@ -282,7 +282,7 @@ in {
         })
       ];
 
-      home.packages = [ cfg.package ];
+      home.packages = if cfg.package != null then [ cfg.package ] else [ ];
```
Comment (teto) [inline modules/programs/waybar.nix:285]: more elegant with `lib.optionals`

Comment (thrix) [reply modules/programs/waybar.nix]: @teto thank you, I hope I got it right, new to nix :)

File: tests/modules/programs/waybar/package-null.nix [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  config = {
+    home.stateVersion = "23.11";
+
+    programs.waybar = {
+      package = null;
+      enable = true;
+    };
+
+    nmt.script = ''
+      assertPathNotExists home-files/.config/waybar/style.css
```
Comment (teto) [inline tests/modules/programs/waybar/package-null.nix:15]: hum shouldn't it check that the executable "wayland" is not available instead ?

Comment (thrix) [reply tests/modules/programs/waybar/package-null.nix:15]: @teto I agree, but I need some help :) I was looking at other tests and did not find if there is already a test does this for other programs. This test would anyway fail on the old code, as `null` was not accepted at all.

Any hints please how to test that waybar package would not be installed?

Comment (thrix) [reply tests/modules/programs/waybar/package-null.nix:15]: @rycee hi, maybe some clues from you, I am a bit new to home-manager and nix :(

Comment (teto) [reply tests/modules/programs/waybar/package-null.nix:15]: just checking `which waybar` return code should be ok ?

Comment (thrix) [reply tests/modules/programs/waybar/package-null.nix:20]: @teto I also had to define the package stub to actually do something :)

It should be resolved, please check!

**Comment by thrix** [comment]
kindly poking for merge, I hope I addressed all the concerns. Now I need to run against my fork :(

**Comment by rycee** [comment]
Sorry, I'm somewhat worried about these `nullOr` package options. The idea is to have configurations be reproducible and bringing in the package from elsewhere goes against that. The traditional way to get around this (if you absolutely insist) is to set the package option to `pkgs.emptyDirectory` and I sort of like that since it is somewhat cumbersome and ugly, which hopefully discourages people from using it 😉

So, basically have you tried `programs.waybar.package = pkgs.emptyDirectory`, if that works well for you then I would prefer that over merging this.

**Comment by thrix** [comment]
> Sorry, I'm somewhat worried about these `nullOr` package options. The idea is to have configurations be reproducible and bringing in the package from elsewhere goes against that. The traditional way to get around this (if you absolutely insist) is to set the package option to `pkgs.emptyDirectory` and I sort of like that since it is somewhat cumbersome and ugly, which hopefully discourages people from using it 😉
> 
> So, basically have you tried `programs.waybar.package = pkgs.emptyDirectory`, if that works well for you then I would prefer that over merging this.

Thank you, that works for me also, so I can drop part of this MR.

Still, I would like to still fix the help if you do not mind:

```
        Waybar package to use. Set to `null` to use the default package.
```

As it is misleading, as a new user I am unable to recognize I should use `pkgs.emptyDirectory` and rather document it explicitely.

And keep the tests to verify.

WDYT @rycee ?

**Comment by thrix** [comment]
@rycee so after more testing, it does not work as well as with my patch, namely I get issues like:

{code}
/tmp/home-manager-build.mEFT4G3Gie/generation/activate: line 391: /nix/store/4d0ix5djms3n2jnjdc58l916cwack1rp-empty-directory/bin/swaymsg: No such file or directory
{code}

I did not see that with my patch :(

**Comment by thrix** [comment]
I decided the proposed solution is enough for me, I can live with swaymsg unavailable for the empty package.

Thanks for you feedback.
